src/EA.Weee.RequestHandlers.Tests.Unit/AatfEvidence/TransferTonnagesValidatorTests.cs

```diff
@@ -0,0 +1,230 @@
+                noteTransferTonnage2
+            };
+            
+            var noteTonnage1 = A.Fake<NoteTonnage>();
+            A.CallTo(() => noteTonnage1.Id).Returns(transferValueTonnageId1);
+            A.CallTo(() => noteTonnage1.Received).Returns(11);
+            A.CallTo(() => noteTonnage1.NoteTransferTonnage).Returns(noteTransferTonnages);
+            A.CallTo(() => evidenceDataAccess.GetTonnageByIds(A<List<Guid>>._))
+                .Returns(new List<NoteTonnage>() { noteTonnage1 });
+
+            //act
+            var exception = await Record.ExceptionAsync(async () => await validator.Validate(transferValues));
+
+            //assert
+            exception.Should().BeNull();
+        }
+
+        [Fact]
+        public async Task Validate_GivenRequestedTonnageIsNoLongerAvailable_InvalidOperationExceptionExpected()
+        {
+            var transferValueTonnageId1 = Guid.NewGuid();
+            var transferValuesTonnage1 = A.Fake<TransferTonnageValue>();
+            A.CallTo(() => transferValuesTonnage1.TransferTonnageId).Returns(transferValueTonnageId1);
+            A.CallTo(() => transferValuesTonnage1.FirstTonnage).Returns(1);
+            var transferValues = new List<TransferTonnageValue>()
+            {
+                transferValuesTonnage1
+            };
+
+            var approvedNote = A.Fake<Note>();
+            A.CallTo(() => approvedNote.Status).Returns(NoteStatus.Approved);
+            var noteTransferTonnage1 = A.Fake<NoteTransferTonnage>();
+            A.CallTo(() => noteTransferTonnage1.Received).Returns(5);
+            A.CallTo(() => noteTransferTonnage1.TransferNote).Returns(approvedNote);
+            var noteTransferTonnage2 = A.Fake<NoteTransferTonnage>();
+            A.CallTo(() => noteTransferTonnage2.Received).Returns(5);
+            A.CallTo(() => noteTransferTonnage2.TransferNote).Returns(approvedNote);
+            var noteTransferTonnages = new List<NoteTransferTonnage>()
+            {
+                noteTransferTonnage1,
+                noteTransferTonnage2
+            };
+
+            var noteTonnage1 = A.Fake<NoteTonnage>();
+            A.CallTo(() => noteTonnage1.Id).Returns(transferValueTonnageId1);
+            A.CallTo(() => noteTonnage1.Received).Returns(10);
+            A.CallTo(() => noteTonnage1.NoteTransferTonnage).Returns(noteTransferTonnages);
+            A.CallTo(() => evidenceDataAccess.GetTonnageByIds(A<List<Guid>>._))
+                .Returns(new List<NoteTonnage>() { noteTonnage1 });
+
+            //act
+            var exception = await Record.ExceptionAsync(async () => await validator.Validate(transferValues));
+
+            //assert
+            exception.Should().BeOfType<InvalidOperationException>();
+        }
+
+        [Theory]
+        [ClassData(typeof(NoteStatusData))]
+        public async Task Validate_GivenRequestedTonnageIsAvailableAsNoteIsNoteTransferIsNotApproved_NoExceptionExpected(NoteStatus status)
```

Comment:
would you mind making it a bit more readable?  the middle bit

Reply:
Tried to clean it up a little